### PR TITLE
Added Compass Resource

### PR DESCRIPTION
**Pull request by 3marproof**
here is a demo

https://youtu.be/ahGx6qvGHnc

### CODE REVIEW

**Comment by ciber96** [comment]
👍

File: compass/meta.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<meta>
+  <info name="Extra Chat" type="script" description="Extra chat functionality and commands" author= "3marproof" version="1.0" />
```
Comment (MissyMelissa) [inline compass/meta.xml]: Why is this called "Extra Chat", and "Extra chat functionality and commands"? Just name it "Compass".

File: compass/compass.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿API.onUpdate.connect(function (sender) {
+
+    if (0.3 < API.getGameplayCamDir().X && 0.3 < API.getGameplayCamDir().Y) {
```
Comment (MissyMelissa) [inline compass/compass.js]: There's absolutely no need to call `API.getGameplayCamDir()` 21 times. This is very slow.

File: compass/compass.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+﻿API.onUpdate.connect(function (sender) {
+
+    if (0.3 < API.getGameplayCamDir().X && 0.3 < API.getGameplayCamDir().Y) {
+
+        API.drawText("NE", 350, 1000, 1, 255, 255, 255, 255, 2, 1, false, true, 0);
```
Comment (MissyMelissa) [inline compass/compass.js]: You can also store the string in a temporary variable instead of duplicating the same `drawText` call every time.

Comment (3marproof) [inline compass/compass.js]: what do you mean, even if i do that, i would still need to do drawText every single time??

File: compass/compass.js (hunk 2 above)
Comment (3marproof) [inline compass/compass.js]: fixed

File: compass/compass.js (hunk 3 above)
Comment (MissyMelissa) [inline compass/compass.js]: Of course, but you'd only need to have the `API.drawText` line once (at the end of the function).

File: compass/compass.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
-﻿API.onUpdate.connect(function (sender) {
+var x;
```
Comment (MissyMelissa) [inline compass/compass.js]: Wait why did you make these global? :P

File: compass/compass.js [hunk 5]
```diff
@@ -1,7 +1,10 @@
-﻿API.onUpdate.connect(function (sender) {
+var x;
+var y;
 
-    var x = API.getGameplayCamDir().X;
-    var y = API.getGameplayCamDir().Y;
+API.onUpdate.connect(function (sender) {
+
+    x = API.getGameplayCamDir().X;
```
Comment (MissyMelissa) [inline compass/compass.js]: You can also reduce this to a single call to `getGameplayCamDir` (it returns a vector)

File: compass/compass.js (hunk 4 above)
Comment (3marproof) [inline compass/compass.js]: idk xD

File: compass/compass.js (hunk 5 above)
Comment (3marproof) [inline compass/compass.js]: even if i do that, i would still need to call X and Y of the variable that i will assign getGameplayCamDir to

Comment (MissyMelissa) [inline compass/compass.js]: Sure, but that's faster than calling `getGameplayCamDir`.

**Comment by MissyMelissa** [comment]
Also, maybe I'm nitpicking, but you have a lot of unnecessary empty lines :P

**Comment by 3marproof** [comment]
ITS CALLED TIDY CODE MMMM K?!

**Comment by MissyMelissa** [comment]
I'd argue that it's causing more noise than tidiness 😊

**Comment by 3marproof** [comment]
anything else i need to do, missing?

**Pull request merged**